swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/TunnelManager.swift

```diff
@@ -378,6 +384,37 @@ public class TunnelManager {
     loop()
   }
 
+  func consumeStopReason() async throws -> String {
```

Comment:
This looks a bit repeated with other callbacks, assuming we can abstract over decoding the response.

Reply:
Agree, I think it would make sense to move the IPC stuff into a dedicated module to keep TunnelManager focused on the actual vpn profile.

Will refactor in #7549